lib/queuePopulator/IngestionProducer.js

```diff
@@ -303,35 +341,36 @@ class IngestionProducer {
             params.KeyMarker = keyMarker;
         }
         // TODO: For testing, I can set MaxKeys
-        const req = this._s3Client.listObjectVersions(params);
-        attachReqUids(req, this.requestLogger);
-        return req.send((err, data) => {
-            if (err) {
+        const command = new ListObjectVersionsCommand(params);
+        attachReqUids(command, this.requestLogger);
+        return this._s3Client.send(command)
+            .then(data => {
+                const {
+                    IsTruncated,
+                    NextKeyMarker,
+                    NextVersionIdMarker,
+                    Versions,
+                    DeleteMarkers
+                } = data;
+
+                const response = {
+                    versionList: [...(Versions || []), ...(DeleteMarkers || [])],
```

Comment:
what is the "value" you get for `Versions` and `DeleteMarkers` ?
if `undefined`, then the fallback to `[]` is not needed, it is perfectly fine to "spread" undefined...